tests/e2e/framework/kubernetes.go

```diff
@@ -0,0 +1,232 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import (
+	"flag"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/glog"
+	"istio.io/istio/tests/e2e/util"
+)
+
+const (
+	yamlSuffix        = ".yaml"
+	mixerHubDefault   = "MIXER_HUB"
+	mixerTagDefault   = "MIXER_TAG"
+	managerHubDefault = "MANAGER_HUB"
+	managerTagDefault = "MANAGER_TAG"
+)
+
+var (
+	// hub and tag is for app template if applicable TODO Find a better way to set default of these two
+	appHubDefault = "gcr.io/istio-testing"
+	appTagDefault = "b121a1e169365865e01a9e6eea066a34a29d9fd1"
+
+	appHub       = flag.String("app_hub", appHubDefault, "app hub")
+	appTag       = flag.String("app_tag", appTagDefault, "app tag")
+	namespace    = flag.String("n", "", "Namespace to use for testing (empty to create/delete temporary one)")
+	mixerImage   = flag.String("mixer", os.Getenv(mixerHubDefault)+"/mixer:"+os.Getenv(mixerTagDefault), "Mixer image")
```

Comment:
About the proxy, I understand they are sharing the same hub and tag with manager, that's why I use manager's default. ProxyHub is kept because it's kind of wired and confused to write "ManagerHub"
ManagerProxy" when talking about proxy